File: app/src/main/java/com/example/realtimechat/instruments/MyActivityResultContract.java
package com.example.realtimechat.instruments;

import android.content.Context;
import android.content.Intent;

import androidx.activity.result.contract.ActivityResultContract;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class MyActivityResultContract extends ActivityResultContract {
    @NonNull
    @Override
    public Intent createIntent(@NonNull Context context, Object input) {
        return null;
    }

    @Override
    public Object parseResult(int resultCode, @Nullable Intent intent) {
        return null;
    }
}

File: app/src/main/java/com/example/realtimechat/instruments/MyActivityResultContract.kt
package com.example.realtimechat.instruments

import android.content.Context
import android.content.Intent
import android.net.Uri
import androidx.activity.result.contract.ActivityResultContract
import com.theartofdev.edmodo.cropper.CropImage
import com.theartofdev.edmodo.cropper.CropImageView

class MyActivityResultContract : ActivityResultContract<Any?, Uri>() {
    override fun createIntent(context: Context, input: Any?): Intent {
        return CropImage.activity()
            .setAspectRatio(1, 1)
            .setRequestedSize(600, 600)
            .setCropShape(CropImageView.CropShape.OVAL)
            .getIntent(context)
    }

    override fun parseResult(resultCode: Int, intent: Intent?): Uri? {
        return CropImage.getActivityResult(intent)?.uri
    }

}